### PR TITLE
rpcserver: fix `recoveryPending` attribute

### PR DESCRIPTION
**Pull request by positiveblue**
The rpcserver cannot have more than one account recovery process running
at once. The server uses a internal attribute (`recoveryPending`) that
is set to true when a new recovery process starts. However, that
attribute was only set to false if the process ended successfully,
leaving the server unable to try more recoveries after a failure.

### CODE REVIEW

**Review by guggero — APPROVED** [review]
Nice fix, thank you!

File: rpcserver.go [hunk 1]
```diff
@@ -1148,10 +1148,17 @@ func (s *rpcServer) RecoverAccounts(ctx context.Context,
 
 	s.recoveryMutex.Lock()
 	if s.recoveryPending {
-		defer s.recoveryMutex.Unlock()
+		s.recoveryMutex.Unlock()
```
Comment (bhandras) [inline rpcserver.go:1151]: nit: alternatively we could just use an atomic int to avoid the (slightly complex) locking logic.

**Review by bhandras — APPROVED** [review]
LGTM :+1:

File: rpcserver.go [hunk 2]
```diff
@@ -1148,10 +1148,17 @@ func (s *rpcServer) RecoverAccounts(ctx context.Context,
...
 		return nil, fmt.Errorf("recovery already in progress")
 	}
+
 	s.recoveryPending = true
+	// Mark the recovery process as done whenever we finish.
```
[the remark this answers is not in the document]
Comment (guggero) [reply rpcserver.go:1157]: Sorry, my comment was a bit inaccurate. But the `s.recoveryPending = true` should still be within the lock.

**Pull request merged**